flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergStreamWriter.java

```diff
@@ -86,6 +86,9 @@ public void endInput() throws IOException {
     // For bounded stream, it may don't enable the checkpoint mechanism so we'd better to emit the
     // remaining completed files to downstream before closing the writer so that we won't miss any
     // of them.
+
+    // Note that if the task is not closed after calling endInput, checkpoint may be triggered again
+    // causing files to be sent repeatedly, so we need to set write to null after the file is sent.
```

Comment:
Nit: “we need to write to null after the last file is sent” -> maybe something more like “the writer is marked as null after the last file is sent to guard against duplicated writes”. It feels weird to me to say “write to null” as we don’t have some special writer that represents `/dev/null` that we swap in. The writer is just set as null.

Also as a heads up, we generally only adjust one Flink version during review — the latest — and then backport the changes to the other Flink versions in a follow up PR. This makes it easier to review as it keeps comments all in the same file, and it also helps ensure the same exact change is backported to the previous versions. There are some cases where this isn’t possible (eg an API breaking change in core would cause compilation to fail if other versions weren’t updated), but otherwise you can just do the latest Flink version and we’ll backport later. Though please make sure that the fix _does_ work on earlier Flink versions if possible 🙂

Reply:
Thank you for your review, I have made changes

Reply:
In this case, since the fix is rather small, I am fine either way: a single PR for all 3 versions or separate PR for backporting to 1.13 and 1.14.

Reply:
> In this case, since the fix is rather small, I am fine either way: a single PR for all 3 versions or separate PR for backporting to 1.13 and 1.14.

At present, it has been modified. Later, the previous version will be changed separately